File: 2015/Python/day-01/solution.py
with open("input.txt", "r") as f:
    data = f.read().strip()

# --- Part 1 ---
floors_up, floors_down = data.count("("), data.count(")")
print("D01P01:", floors_up + (-1 * floors_down))

# --- Part 2 ---
floor = 0
for i, c in enumerate(list(data)):
    if c == "(":
        floor += 1
    elif c == ")":
        floor -= 1
    if floor == -1:
        print("D01P02:", i + 1)
        break

File: 2015/Go/day-01/solution.go
package main

import (
	"fmt"
	"io/ioutil"
  "strings"
)

func doP1(data string) uint16 {
  return uint16(strings.Count(data, "(")) - uint16(strings.Count(data, ")"));
}

func doP2(data string) uint16 {
  var f int16 = 0;
  for i := 0; i < len(data); i++ {
    c := string(data[i]);
    if c == "(" {
      f++;
    } else {
      f--;
    }
    if f == -1 {
      return uint16(i) + 1;
    }
  }
  return 0;
}

func main() {
  body, err := ioutil.ReadFile("input.txt");
  if err != nil {
    panic("Could not real file!");
  }
  var data = string(body);
  p1 := doP1(data);
  fmt.Printf("Day 01, Part 01: %d\n", p1);
  p2 := doP2(data);
  fmt.Printf("Day 01, Part 02: %d\n", p2);
}
